kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js

```diff
@@ -164,7 +150,16 @@ define([
         }
 
         function setModelValue(value) {
+            // This might be a bit of a cheat. I'm not sure what'll happen if we just
+            // remove the current value from disabled values before updating.
+            // WE'LL SEE!
```

Comment:
😱

Reply:
For the record, it works fine. This is just in there to force the update early. The order of events when clicking the little "clear selection x" button seems to be (before this bit):
1. delete selection
2. set model value 
3. open dropdown
4. validate new value
5. get new list of disabled / enabled values from widget controller

So it was deleting the value, showing the error that you need a new value at the same time as opening the dropdown, and not actually showing that the recently deleted value is now available. You'd have to close the dropdown first so the select2-rendered bits would get the update, THEN it would become available again. It was really annoying, especially since we know that once the item is cleared or changed, it should instantly become available as a selection again.

This **might** have some longer term issues, but was an easier fix than figuring out how to make select2 not open the dropdown when hitting the clear button.